apps/financial-aid/web-osk/src/components/Layout/Layout.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { ReactNode, useContext, useEffect } from 'react'
+import { Box, GridContainer } from '@island.is/island-ui/core'
+
+import * as styles from './Layout.treat'
+
+import {
+  Login,
+  HasApplied,
+  FormLayout,
+} from '@island.is/financial-aid-web/osk/src/components'
+
+import { UserContext } from '@island.is/financial-aid-web/osk/src/components/UserProvider/UserProvider'
+import cn from 'classnames'
+interface Props {
+  children: ReactNode
+}
+
+const Layout = ({ children }: Props) => {
+  const { isAuthenticated, user } = useContext(UserContext)
+
+  useEffect(() => {
+    document.title = 'Umsókn um fjárhagsaðstoð'
+  }, [])
+
+  if (!isAuthenticated) {
+    return <Login headline="Skráðu þig inn" />
+  }
+  if (!user) {
+    return null
+  }
+
+  return (
+    <Box
+      paddingY={[3, 3, 3, 6]}
+      background="purple100"
+      className={styles.processContainer}
+    >
+      <GridContainer
+        className={cn({
+          [`${styles.gridContainer}`]: !user.currentApplication,
+        })}
+      >
+        {user.currentApplication ? (
+          <HasApplied />
+        ) : (
+          <FormLayout>{children}</FormLayout>
+        )}
```

Comment:
Don't have a suggestion but can you think of any way of having this ternarny elsewhere?

The first thing that pops into my mind is to return a redirect to hasApplied if this happens.

Reply:
I don't have any at the moment, I tried Redirect it didn't work, so if you find good solution I am open to that. However I dont think this is a horrible solution, the user should use his status page  link to see his application, I feel like this is just in case.